pkg/segment/segexecution.go

```diff
@@ -335,6 +340,9 @@ func HelperQueryArithmeticAndLogical(queryOp *structs.QueryArithmetic, resMap ma
 				for rGroupId := range resultRHS.Results {
 					matchingLabelValTorightGroupID[matchingLabelValStr] = rGroupId
 				}
+			} else {
+				// ManyToMany, Set this to false so that only the matching label sets are processed.
+				*opLabelsDoNotNeedToMatch = false
 			}
```

Comment:
This might be fine; however, I am a little concerned that this is being set so deeply in the call stack (`ProcessQueryArithmeticAndLogical` gets called with a boolean, which passes a pointer to that bool to `processQueryArithmeticNodeOp`, which calls `HelperQueryArithmeticAndLogical`, which sets the boolean here). I think this will make the code harder to reason about if we have to make changes later

What do you think about this: before calling `ProcessQueryArithmeticAndLogical`, we do some extra work to find the correct value of `opLabelsDoNotNeedToMatch`, then pass it as a boolean (not a pointer) through the whole call stack.

Is that feasible? Do you think it would help make the code easier to understand?

Reply:
Yes, I have updated the code to reflect your comment.